eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/SystemEventTypeMappings.java

```diff
@@ -0,0 +1,95 @@
+import com.microsoft.azure.eventgrid.models.StorageBlobCreatedEventData;
+import com.microsoft.azure.eventgrid.models.StorageBlobDeletedEventData;
+import com.microsoft.azure.management.apigeneration.Beta;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Mapping of system event type name to corresponding type of the Java model.
+ */
+@Beta
+final class SystemEventTypeMappings {
+    /**
+     * The map containing system eventType to Java model type mapping.
+     */
+    private static Map<String, Type> systemEventMappings;
+
+    static {
+        systemEventMappings = new HashMap<>(); // key: eventType, Value:eventDataType
+        // ContainerRegistry events
+        systemEventMappings.put(SystemEventTypes.CONTAINER_REGISTRY_IMAGE_PUSHED_EVENT.toLowerCase(), ContainerRegistryImagePushedEventData.class);
+        systemEventMappings.put(SystemEventTypes.CONTAINER_REGISTRY_IMAGE_DELETED_EVENT.toLowerCase(), ContainerRegistryImageDeletedEventData.class);
+        // Device events
+        systemEventMappings.put(SystemEventTypes.IOT_HUB_DEVICE_CREATED_EVENT.toLowerCase(), IotHubDeviceCreatedEventData.class);
+        systemEventMappings.put(SystemEventTypes.IOT_HUB_DEVICE_DELETED_EVENT.toLowerCase(), IotHubDeviceDeletedEventData.class);
+        // EventGrid events
+        // TODO: Enable this once SDK is refreshed
+        // systemEventMappings.put(SystemEventTypes.EVENT_GRID_SUBSCRIPTION_VALIDATION_EVENT.toLowerCase(), SubscriptionValidationEventData.class);
+        // systemEventMappings.put(SystemEventTypes.EVENT_GRID_SUBSCRIPTION_DELETED_EVENT.toLowerCase(), SubscriptionDeletedEventData.class);
+        // Event Hub Events
+        systemEventMappings.put(SystemEventTypes.EVENT_HUB_CAPTURE_FILE_CREATED_EVENT.toLowerCase(), EventHubCaptureFileCreatedEventData.class);
+        // Media Services events
+        // TODO: Enable this once SDK is refreshed
+        // systemEventMappings.put(SystemEventTypes.MEDIA_JOB_STATE_CHANGE_EVENT.toLowerCase(), MediaJobStateChangeEventData.class);
+        // Resource Manager (Azure Subscription/Resource Group) events
+        systemEventMappings.put(SystemEventTypes.RESOURCE_WRITE_SUCCESS_EVENT.toLowerCase(), ResourceWriteSuccessData.class);
+        systemEventMappings.put(SystemEventTypes.RESOURCE_WRITE_FAILURE_EVENT.toLowerCase(), ResourceWriteFailureData.class);
+        systemEventMappings.put(SystemEventTypes.RESOURCE_WRITE_CANCEL_EVENT.toLowerCase(), ResourceWriteCancelData.class);
+        systemEventMappings.put(SystemEventTypes.RESOURCE_DELETE_SUCCESS_EVENT.toLowerCase(), ResourceDeleteSuccessData.class);
+        systemEventMappings.put(SystemEventTypes.RESOURCE_DELETE_FAILURE_EVENT.toLowerCase(), ResourceDeleteFailureData.class);
+        systemEventMappings.put(SystemEventTypes.RESOURCE_DELETE_CANCEL_EVENT.toLowerCase(), ResourceDeleteCancelData.class);
```

Comment:
There are two more events that need to be added to this category after merging of the related PR.

  { EventTypes.ResourceActionFailureEvent, typeof(ResourceActionFailureData) },
  { EventTypes.ResourceActionCancelEvent, typeof(ResourceActionCancelData) },

Reply:
and 
{ EventTypes.ResourceActionSuccessEvent, typeof(ResourceActionSuccessData) },

Reply:
Thanks, you are right, yes there are three of them.